src/requester.ts

```diff
@@ -0,0 +1,89 @@
+import bsv from 'bsv';
+import { AuthHeader, setSignature } from './authentication';
+import { Logger } from './logger';
+import { ErrorNoAdminKey, ErrorResponse } from './errors';
+
+export class Requester {
+  private logger: Logger;
+
+  private adminKey?: bsv.HDPrivateKey;
+  private signingKey?: bsv.HDPrivateKey | bsv.PrivateKey;
+  private xPubString?: string;
+  private url: string;
+
+  static CreateSigningRequester(
+    logger: Logger,
+    url: string,
+    signingKey?: bsv.HDPrivateKey | bsv.PrivateKey,
+    adminKey?: bsv.HDPrivateKey,
+  ) {
+    const requester = new Requester(logger, url);
+    requester.signingKey = signingKey;
+    requester.adminKey = adminKey;
+    return requester;
+  }
+
+  static CreateXPubRequester(logger: Logger, url: string, xPubString: string) {
+    const requester = new Requester(logger, url);
+    requester.xPubString = xPubString;
+    return requester;
+  }
+
+  private constructor(logger: Logger, url: string) {
+    this.logger = logger;
+    this.url = url.endsWith('/') ? url : url + '/'; //make sure the url ends with a '/'
+  }
+
+  async AdminRequest(path: string, method: string = 'GET', payload: any = null): Promise<any> {
+    if (!this.adminKey) {
+      throw new ErrorNoAdminKey();
+    }
+    this.logger.debug('Making AdminRequest', path);
+    return this.makeRequest(path, method, payload, this.adminKey);
+  }
+
+  async Request(path: string, method: string = 'GET', payload: any = null): Promise<any> {
+    this.logger.debug('Making Request', path);
+    return this.makeRequest(path, method, payload, this.signingKey);
+  }
+
+  private async makeRequest(
+    path: string,
+    method: string,
```

Comment:
IMHO method here and above should be typed string like:
```
type HttpMethod = 'GET' | 'POST' ...
```

Reply:
Defined `type HttpMethod`